plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -766,6 +779,57 @@ export class EventsProcessor {
         }
     }
 
+    private tryStoreSessionRecordingToObjectStorage(
+        timestamp: DateTime,
+        session_id: string,
+        snapshot_data: Record<any, any>,
+        team_id: number
+    ): Record<any, any> {
+        // As we don't want to store the session recording payload in ClickHouse,
+        // let's intercept the event, parse the metadata and store the data in
+        // our object storage system.
+
+        if (!this.objectStorage.isEnabled) {
+            return snapshot_data
+        }
+
+        const teamIsOnSessionRecordingToS3AllowList =
+            this.objectStorage.sessionRecordingAllowList === 'all' ||
+            this.objectStorage.sessionRecordingAllowList.includes(team_id)
+
+        if (!teamIsOnSessionRecordingToS3AllowList) {
+            return snapshot_data
+        }
+
+        const dateKey = castTimestampOrNow(timestamp, TimestampFormat.DateOnly)
+        const object_storage_path = `${OBJECT_STORAGE_SESSION_RECORDING_FOLDER}/${dateKey}/${session_id}/${snapshot_data.chunk_id}/${snapshot_data.chunk_index}`
+        const params = { Bucket: OBJECT_STORAGE_BUCKET, Key: object_storage_path, Body: snapshot_data.data }
+
+        const tags = {
+            team_id: team_id.toString(),
+            session_id,
+        }
+
+        const storageWriteTimer = new Date()
+
+        this.objectStorage.putObject(params, (err: any, resp: any) => {
+            if (err) {
+                console.error(err)
+                this.pluginsServer.statsd?.increment('session_data.storage_upload.error', tags)
+            } else {
+                this.pluginsServer.statsd?.increment('session_data.storage_upload.success', tags)
+            }
+        })
+
+        this.pluginsServer.statsd?.timing('session_data.storage_upload.timing', storageWriteTimer, tags)
```

Comment:
This timer will always be zero since we call it just after queueing a callback.

Reply:
![homer-hedge](https://user-images.githubusercontent.com/984817/169995057-c379a220-69bb-43cf-87c1-2f5fbb510028.gif)